hibernate-core/src/main/java/org/hibernate/id/Configurable.java

```diff
@@ -21,6 +22,13 @@
  * @author Steve Ebersole
  */
 public interface Configurable {
+	/**
+	 * @deprecated Use {@link #configure(Type, Properties, GeneratorCreationContext)} instead
+	 */
+	@Deprecated( since = "7.0", forRemoval = true )
+	default void configure(Type type, Properties parameters, ServiceRegistry serviceRegistry) throws MappingException {
```

Comment:
I believe the `Type` not available from the `GeneratorCreationContext` via `context.getProperty().getType()`, so I would not repeat it here. I would make the signature:

    void configure(ServiceRegistry serviceRegistry, Properties parameters)

WDYT?